src/master-playlist-controller.js

```diff
@@ -273,9 +273,26 @@ export class MasterPlaylistController extends videojs.EventTarget {
         // playlist's `segments` list is already available, a media playlist won't be
         // requested, and loadedplaylist won't fire again, so the playlist handler must be
         // called on its own here.
-        if (this.sourceType_ === 'vhs-json' && this.initialMedia_.segments) {
-          this.handleUpdatedMediaPlaylist(this.initialMedia_);
+        if (this.sourceType_ !== 'vhs-json' && !this.initialMedia_.segments) {
+          return;
         }
+        updatedPlaylist = this.initialMedia_;
```

Comment:
We have to check muxer support for codecs, and this code was just doing the same thing as the code below.